sklearn/exceptions.py

```diff
@@ -0,0 +1,33 @@
+"""
+The :mod:`sklearn.exceptions` module includes all custom warnings and error
+classes used across scikit-learn.
+"""
+
+
+__all__ = ['NotFittedError', 'FitFailedWarning', 'ChangedBehaviorWarning',
+           'DataConversionWarning', 'EfficiencyWarning']
+
+
+class NotFittedError(ValueError, AttributeError):
+    """Exception class to raise if estimator is used before fitting
+
+    This class inherits from both ValueError and AttributeError to help with
+    exception handling and backward compatibility.
+    """
+
+
+class FitFailedWarning(RuntimeWarning):
```

Comment:
When taken out of context of a particular module, these need docstrings to explain when they should be used or expected. It may even be appropriate to add these to `doc/modules/classes.rst` as part of the public API.